includes/rest-api/class-sensei-rest-api-setup-wizard-controller.php

```diff
@@ -298,6 +300,73 @@ public function register_get_data_route() {
 		);
 	}
 
+	/**
+	 * Register setup wizard user data option in the settings REST API endpoint.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function register_setup_wizard_settings() {
+
+		register_setting(
+			'options',
+			Sensei_Setup_Wizard::USER_DATA_OPTION,
+			[
+				'type'         => 'object',
+				'show_in_rest' => [
+					'schema' => [
+						'properties' => [
+							'purpose' => [
+								'type'       => 'object',
+								'properties' => [
+									'selected' => [
+										'required' => true,
+										'type'     => 'array',
+										'items'    => [
+											'type' => 'string',
+											'enum' => self::PURPOSES,
+										],
+									],
+									'other'    => [
+										'required' => true,
+										'type'     => 'string',
+									],
+								],
+							],
+						],
+					],
+				],
+			]
+		);
+
+		add_filter( 'rest_pre_update_setting', 'update_setup_wizard_settings', 10, 3 );
+	}
+
+	/**
+	 * Update setup wizard user data option when it's set via the REST API.
+	 * Ensures the option is complete with the default values if not set.
+	 *
+	 * @hooked rest_pre_update_setting
+	 * @access private
+	 *
+	 * @param bool   $updated Whether to override the default behavior for updating the
+	 *                        value of a setting.
+	 * @param string $name   Setting name (as shown in REST API responses).
+	 * @param mixed  $value  Updated setting value.
+	 */
+	public function update_setup_wizard_settings( $updated, $name, $value ) {
+		if ( Sensei_Setup_Wizard::USER_DATA_OPTION !== $name ) {
+			return $updated;
+		}
+		if ( ! empty( $settings ) ) {
```

Comment:
$settings is not defined.

Reply:
Ah that's why it didn't fix the issue :(